mp/src/game/client/neo/ui/neo_hud_friendly_marker.cpp

```diff
@@ -66,87 +69,103 @@ CNEOHud_FriendlyMarker::CNEOHud_FriendlyMarker(const char* pElemName, vgui::Pane
+void CNEOHud_FriendlyMarker::DrawNeoHudElement()
 {
-	int x, y;
+	if (!ShouldDraw())
+	{
+		return;
+	}
 	const float scale = neo_friendly_marker_hud_scale_factor.GetFloat();
-	const float heightOffset = 48.0f;
-	Vector pos;
-
-	auto localPlayer = C_NEO_Player::GetLocalNEOPlayer();
-
-	Assert(localPlayer->m_rvFriendlyPlayerPositions.Count() == MAX_PLAYERS);
 
-	COMPILE_TIME_ASSERT(ARRAYSIZE(m_x0) == MAX_PLAYERS);
-	COMPILE_TIME_ASSERT(ARRAYSIZE(m_x1) == MAX_PLAYERS);
-	COMPILE_TIME_ASSERT(ARRAYSIZE(m_y0) == MAX_PLAYERS);
-	COMPILE_TIME_ASSERT(ARRAYSIZE(m_y1) == MAX_PLAYERS);
+	m_iMarkerWidth = (m_iMarkerTexWidth * 0.5f) * scale;
+	m_iMarkerHeight = (m_iMarkerTexHeight * 0.5f) * scale;	
 
-	for (int i = 0; i < MAX_PLAYERS; ++i)
+	auto localPlayer = C_NEO_Player::GetLocalNEOPlayer();
+	auto team = localPlayer->GetTeam();
+	m_IsSpectator = team->GetTeamNumber() == TEAM_SPECTATOR;
+	auto memberCount = team->GetNumPlayers();
+	
+	if(m_IsSpectator)
 	{
-		if ((localPlayer->entindex() == i + 1) || // Skip self
-			(localPlayer->m_rvFriendlyPlayerPositions[i] == vec3_origin)) // Skip unused positions
+		for(int t = 0; t < 2; ++t)
 		{
-			continue;
+			team = GetGlobalTeam(TEAM_SPECTATOR + t + 1);
+			memberCount = team->GetNumPlayers();
+			for (int i = 0; i < memberCount; ++i)
+			{
+				auto player = team->GetPlayer(i);
+				if(player && player->IsAlive())
+				{
+					DrawPlayer(GetTeamColour(team->GetTeamNumber()), player);
+				}		
+			}
 		}
-
-		pos = Vector(
-			localPlayer->m_rvFriendlyPlayerPositions[i].x,
-			localPlayer->m_rvFriendlyPlayerPositions[i].y,
-			localPlayer->m_rvFriendlyPlayerPositions[i].z + heightOffset);
-
-		if (GetVectorInScreenSpace(pos, x, y))
+	} else
+	{
+		for (int i = 0; i < memberCount; ++i)
 		{
-			m_x0[i] = RoundFloatToInt((x - ((m_iMarkerTexWidth * 0.5f) * scale)));
-			m_x1[i] = RoundFloatToInt((x + ((m_iMarkerTexWidth * 0.5f) * scale)));
-			m_y0[i] = RoundFloatToInt((y - ((m_iMarkerTexHeight * 0.5f) * scale)));
-			m_y1[i] = RoundFloatToInt((y + ((m_iMarkerTexHeight * 0.5f) * scale)));
+			auto player = team->GetPlayer(i);
+			if(player && localPlayer->entindex() != player->entindex() && player->IsAlive())
+			{
+				DrawPlayer(GetTeamColour(team->GetTeamNumber()), player);
+			}		
 		}
 	}
 }
 
-void CNEOHud_FriendlyMarker::DrawNeoHudElement()
+void CNEOHud_FriendlyMarker::DrawPlayer(Color teamColor, C_BasePlayer* player) const
 {
-	if (!ShouldDraw())
-	{
-		return;
-	}
-
-	SetFgColor(Color(0, 0, 0, 0));
-	SetBgColor(Color(0, 0, 0, 0));
-
-	const Color teamColor = (C_NEO_Player::GetLocalNEOPlayer()->GetTeamNumber() == TEAM_NSF) ? COLOR_NSF : COLOR_JINRAI;
-
-	surface()->DrawSetTextFont(m_hFont);
-	surface()->DrawSetTextColor(teamColor);
-
-	surface()->DrawSetColor(teamColor);
-	surface()->DrawSetTexture(m_hTex);
-
-	auto localPlayer = C_NEO_Player::GetLocalNEOPlayer();
-
-	for (int i = 0; i < MAX_PLAYERS; ++i)
+	int x, y;
+	constexpr float heightOffset = 48.0f;
```

Comment:
Until we've moved off the very old steam-runtime on Linux, I'm not sure if constexpr will compile on GCC 4.6. If not, change this to `static const float`. Also applies to: `maxNameLenght` constexpr usage (and a typo)